torch/ao/quantization/pt2e/representation/rewrite.py

```diff
@@ -93,6 +96,41 @@ def _reference_quantized_add(
     out_i8 = torch.ops.aten.clamp(out_i32, quant_min, quant_max).to(torch.int8)
     return out_i8
 
+_QUANTIZED_MAX_POOL2D_EXAMPLE_INPUTS = (
+    torch.randint(-128, 127, (1, 3, 3, 3), dtype=torch.int8),
+    torch.randn(1, dtype=torch.float),
+    torch.zeros(1, dtype=torch.int),
+    torch.randn(1, dtype=torch.float),
+    torch.zeros(1, dtype=torch.int),
+    torch.tensor([-128], dtype=torch.int),
+    torch.tensor([127], dtype=torch.int),
+)
+
+def _qdq_quantized_max_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_zero_point, quant_min, quant_max):
+    kernel_size = 1
+    stride = 1
+    padding = 0
+    dilation = 1
+    ceil_mode = False
+    x_fp32 = torch.ops.quantized_decomposed.dequantize_per_tensor(x_i8, x_scale, x_zero_point, quant_min, quant_max, torch.int8)
+    out_fp32, _ = torch.ops.aten.max_pool2d_with_indices.default(x_fp32, kernel_size, stride, padding, dilation, ceil_mode)
+    out_i8 = torch.ops.quantized_decomposed.quantize_per_tensor(out_fp32, out_scale, out_zero_point, quant_min, quant_max, torch.int8)
+    return out_i8
+
+def _reference_quantized_max_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_zero_point, quant_min, quant_max):
+    kernel_size = 1
+    stride = 1
+    padding = 0
+    dilation = 1
+    ceil_mode = False
+    acc_i8, _ = torch.ops.aten.max_pool2d_with_indices.default(x_i8, kernel_size, stride, padding, dilation, ceil_mode)
+    acc_i32 = acc_i8.to(torch.int32)
+    # TODO: use mul.Scalar, need to change how we handle literal args
+    output_i32 = out_dtype(torch.ops.aten.mul.Tensor, torch.int32, acc_i32, (x_scale / out_scale))
+    output_i8 = output_i32 - (x_zero_point * x_scale / out_scale + out_zero_point)
+    output_i8 = output_i8.to(torch.int8)
```

Comment:
I think this should really be simplified with 
```
# these can be dtype specialized by doing zp subtraction in int16 when quantized dtype is int8 and int32 when quantized dtype is int16
x_i32 = x_i8.to(torch.int32) - x_zero_point.to(torch.int32)
acc_i32, _ = torch.ops.aten.max_pool2d_with_indices.default(x_i8, kernel_size, stride, padding, dilation, ceil_mode)
output_i32 = out_dtype(torch.ops.aten.mul.Tensor, torch.int32, acc_i32, (x_scale / out_scale))
output_i8 = output_i32 - x_zero_point
```
then as a nit requant logic can be factored out.

Reply:
actually this will make input to max_pool2d op int16/int32 right? then we are not doing maxpool2d operation in int8

Reply:
I think thats fine because in quantized matmul we do matmul in int16 due to zero point subtraction. But you are also right if backends are actually lowering exactly this representation. That is why the representation is really backend tied. We providing one here should mostly be for conveying numerics.

Reply:
OK, maybe I can use int16 here as well?